### PR TITLE
Import master encryption key in `iotedge config import`

### PR DESCRIPTION
**Pull request by arsing**
The workload API exposes an encrypt and decrypt API for modules that derives
module-specific keys using the module ID and a master encryption key.

This change adds an `imported_master_encryption_key` property to
the super-config that gets set to the path of
the IoT Edge 1.1 master encryption key by `iotedge config import`.
`iotedge config apply` then copies the contents of this file to a new file
under `/var/secrets/aziot/keyd` and preloads it as the master encryption key
in the keyd config with the same key ID that edged uses.

This change also changes edged to use the master encryption key directly
to encrypt/decrypt secrets via the workload API. This was also the behavior of
IoT Edge v1.1 and earlier, but for v1.2 we changed the code to use
the master encryption key as a derivation key and
derive per-module encryption keys for this. With this commit edged uses
the master encryption key directly again.

### CODE REVIEW

**Review by massand — APPROVED** [review]

File: edgelet/iotedge/src/config/import/mod.rs [hunk 1]
```diff
@@ -69,7 +69,21 @@ To reconfigure IoT Edge, run:
         .into());
     }
 
-    let config = execute_inner(old_config_file)?;
+    let old_master_encryption_key_path = {
+        // libiothsm derived the filenames for encryption keys from their alias (see `normalize_alias_file_path`)
+        // so the name of the master encryption key can be hard-coded.
+        const OLD_MASTER_ENCRYPTION_KEY_PATH: &str =
+            "/var/lib/iotedge/hsm/enc_keys/edgelet-masterWt5mT2xpO72EPKlt2Tt0Sq4uJCrMvfl2rzzKRB3pnyo_.enc.key";
```
Comment (varunpuranik) [inline edgelet/iotedge/src/config/import/mod.rs:76]: where is this coming from?

Comment (Arnavion) [reply edgelet/iotedge/src/config/import/mod.rs:76]: See the comment.

Comment (arsing) [reply edgelet/iotedge/src/config/import/mod.rs:76]: See the comment.

**Review by varunpuranik — DISMISSED** [review]

**Comment by arsing** [comment]
@massand The third commit ( https://github.com/Azure/iotedge/pull/4697/commits/54233bb3bedc8911ad292f5830ca1c23118deaf9 ) reverts the edged to 1.1's behavior of using the master encryption key itself for the workload API instead of deriving per-module keys. The module ID + generation ID is still used for the AAD as before.

Any concerns?

**Comment by massand** [comment]
> @massand The third commit ( [54233bb](https://github.com/Azure/iotedge/commit/54233bb3bedc8911ad292f5830ca1c23118deaf9) ) reverts the edged to 1.1's behavior of using the master encryption key itself for the workload API instead of deriving per-module keys. The module ID + generation ID is still used for the AAD as before.
> 
> Any concerns?

I am ok with this change for now, since it avoids the need for migration. We can assess the security of this approach with Crypto review team in the near future.

**Review by massand — APPROVED** [review]

**Review by varunpuranik — APPROVED** [review]

File: edgelet/iotedge/src/config/apply.rs [hunk 2]
```diff
@@ -205,6 +222,28 @@ fn execute_inner(
         ],
     });
 
+    let preloaded_master_encryption_key_bytes =
```
Comment (daprilik) [inline edgelet/iotedge/src/config/apply.rs:225]: for clarity's sake, can you wrap this assignment in some braces?

File: edgelet/iotedge/src/config/apply.rs [hunk 3]
```diff
@@ -419,13 +459,39 @@ mod tests {
                     Err(err) => panic!("could not read device-id file: {}", err),
                 };
 
+            let expected_preloaded_master_encryption_key_bytes =
```
Comment (daprilik) [inline edgelet/iotedge/src/config/apply.rs:462]: ditto

File: edgelet/iotedge/src/config/apply.rs (hunk 2 above)
Comment (arsing) [reply edgelet/iotedge/src/config/apply.rs:225]: Done.

File: edgelet/iotedge/src/config/apply.rs (hunk 3 above)
Comment (arsing) [reply edgelet/iotedge/src/config/apply.rs:462]: Done.

**Review by daprilik — APPROVED** [review]

**Pull request merged**